backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/dsl/impl/ElasticsearchStringIndexFieldTypeOptionsStep.java

```diff
@@ -187,6 +188,7 @@ public ElasticsearchStringIndexFieldTypeOptionsStep aggregable(Aggregable aggreg
 			builder.queryElementFactory( PredicateTypeKeys.PHRASE, new ElasticsearchTextPhrasePredicate.Factory() );
 			builder.queryElementFactory( PredicateTypeKeys.WILDCARD, new ElasticsearchTextWildcardPredicate.Factory() );
 			builder.queryElementFactory( PredicateTypeKeys.REGEXP, new ElasticsearchTextRegexpPredicate.Factory() );
+			builder.queryElementFactory( PredicateTypeKeys.TERMS, new ElasticsearchTermsPredicate.Factory( codec ) );
```

Comment:
A diamond operator is missing here and in a few other places.

Please have a look at the sonar report: https://sonarcloud.io/project/issues?id=org.hibernate.search%3Ahibernate-search-parent&open=AXhowV6ZNXwoQfKpUHbp&pullRequest=2532&resolved=false&types=CODE_SMELL

Reply:
The other places are about the `Builder` parameter of the constructors of the predicates.
I'm fixing it, but I think we have the same issue on all predicates (both on ES and Lucene).